pre_commit/commands/install_uninstall.py

```diff
@@ -70,6 +70,22 @@ def _install_hook_script(
 ) -> None:
     hook_path, legacy_path = _hook_paths(hook_type, git_dir=git_dir)
 
+    # If the hooks directory is a symlink we need to be careful
+    if os.path.islink(os.path.dirname(hook_path)):
+        git_dir = git_dir if git_dir is not None else git.get_git_common_dir()
+        # If the hooks directory links to a directory outside the
+        # git repo we shouldn't try to mess with it
+        if os.path.commonpath(
```

Comment:
This would be a bit cleaner using pathlib:
```python3
Path.absolute(Path(hook_path)).is_relative_to(git_dir)
```
but I'm sure you have good reasons not to use pathlib.